.github/workflows/main.yml

```diff
@@ -2,39 +2,62 @@ name: CI Workflow
 
 on:
   pull_request:
-    branches : [ develop ]
-  
-  workflow_dispatch: 
-  
+    branches: [develop]
+
+  workflow_dispatch:
+
   push:
 
 jobs:
   python-tests:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v1
-    - name: Set up Python
-      uses: actions/setup-python@v1
-      with:
-        python-version: 3.8
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        pip install -r requirements/dev.txt
-        pip install pytest-django
-    - name: Run migrations
-      run: python project/manage.py migrate
-    - name: Run tests
-      env:
-        CIVIWIKI_LOCAL_NAME : True
-        DJANGO_SETTINGS_MODULE: core.settings
-      run: |
-        cd project
-        pytest -p no:warning
-    - name: Lint
-      uses: py-actions/flake8@v1
-      with:
-        ignore: "E203,W503"
-        max-line-length: "88"
-        exclude: "**/migrations/*.py"
-
+      - name: Check out repository
+        uses: actions/checkout@v2
+      - name: Set up python
+        id: setup-python
+        uses: actions/setup-python@v2
+        with:
+          python-version: "3.10"
+      #----------------------------------------------
+      #  -----  install & configure poetry  -----
+      #----------------------------------------------
```

Comment:
You can remove the comment because the job is self-explanatory whereas your comments regarding cached dependencies can stay.

Reply:
Done